sentry_sdk/integrations/chalice.py

```diff
@@ -0,0 +1,182 @@
+            if not isinstance(response, Response):
+                response = Response(body=response)
+            app._validate_response(response)
+        except ChaliceViewError as e:
+            # Any chalice view error should propagate.  These
+            # get mapped to various HTTP status codes in API Gateway.
+            response = Response(
+                body={'Code': e.__class__.__name__, 'Message': str(e)},
+                status_code=e.STATUS_CODE,
+            )
+            hub.flush()
+        except Exception:
+            with capture_internal_exceptions():
+                scope.transaction = app.lambda_context.function_name
+                scope.add_event_processor(
+                    _make_request_event_processor(
+                        app.current_request, app.lambda_context
+                    )
+                )
+            exc_info = sys.exc_info()
+            event, hint = event_from_exception(
+                exc_info,
+                client_options=client.options,
+                mechanism={"type": "chalice", "handled": False},
+            )
+            hub.capture_event(event, hint=hint)
+            hub.flush()
+            headers = {}
+            app.log.error(
+                "Caught exception for %s", view_function, exc_info=True
+            )
+            if app.debug:
+                # If the user has turned on debug mode,
+                # we'll let the original exception propagate so
+                # they get more information about what went wrong.
+                stack_trace = ''.join(traceback.format_exc())
+                body = stack_trace
+                headers['Content-Type'] = 'text/plain'
+            else:
+                body = {
+                    'Code': 'InternalServerError',
+                    'Message': 'An internal server error occurred.',
+                }
+            response = Response(body=body, headers=headers, status_code=500)
+    return response
+
+
+class ChaliceIntegration(Integration):
+    identifier = "chalice"
+
+    @staticmethod
+    def setup_once():
+        # for @app.route()
+        Chalice._get_view_function_response = _get_view_function_response
+        # for everything else (like events)
+        AwsLambdaIntegration.setup_once()
+
+
+def _make_request_event_processor(current_request, lambda_context):
+    # type: (Any, Any) -> EventProcessor
+    start_time = datetime.now()
+
+    def event_processor(event, hint, start_time=start_time):
+        # type: (Event, Hint, datetime) -> Optional[Event]
+        extra = event.setdefault("extra", {})
+        extra["lambda"] = {
+            "function_name": lambda_context.function_name,
+            "function_version": lambda_context.function_version,
+            "invoked_function_arn": lambda_context.invoked_function_arn,
+            "aws_request_id": lambda_context.aws_request_id,
+        }
+
+        extra["cloudwatch logs"] = {
```

Comment:
This should not be necessary if the aws lambda integration is enabled

Reply:
this is now addressed by the change mentioned in the comment above https://github.com/getsentry/sentry-python/pull/779#discussion_r470299614

Reply:
Can we still deduplicate the logic of adding cloudwatch logs in both chalice and aws lambda? I.e. make sure that we write the same key in extra (so we don't accidentally send the same info twice)

Ideally you'd call the event processor from aws lambda in here.

Reply:
This is resolved adding the event processor from aws lambda as you suggested me